### PR TITLE
Remove credential subject id

### PR DESCRIPTION
**Pull request by vineetpant**
## Description
This Mr removes credential subject id from Credential and other Credential related structs (offer, credential_request e.t.c)

## Details

- Remove credential_subject.id from structs
- Adjust vade functions and other functions payload
- Adjust test cases
- Create new test data

### CODE REVIEW

File: src/application/issuer.rs [hunk 1]
```diff
@@ -554,11 +549,10 @@ mod tests {
 
     #[test]
     fn can_offer_credential() -> Result<(), Box<dyn Error>> {
-        let proposal: CredentialProposal = serde_json::from_str(&EXAMPLE_CREDENTIAL_PROPOSAL)?;
-        let offer = Issuer::offer_credential(proposal.subject.as_deref(), &ISSUER_DID, 1)?;
+        let _proposal: CredentialProposal = serde_json::from_str(&EXAMPLE_CREDENTIAL_PROPOSAL)?;
```
Comment (wulfraem) [inline src/application/issuer.rs:552]: Hm, do we need the `_proposal` here and in the following sections?

File: src/application/prover.rs [hunk 2]
```diff
@@ -57,14 +57,9 @@ impl Prover {
     ///
     /// # Returns
     /// * `CredentialProposal` - The message to be sent to an issuer
-    pub fn propose_credential(
-        issuer_did: &str,
-        subject_did: Option<&str>,
-        schema_did: &str,
-    ) -> CredentialProposal {
+    pub fn propose_credential(issuer_did: &str, schema_did: &str) -> CredentialProposal {
```
Comment (wulfraem) [inline src/application/prover.rs:60]: Can you update the function documentation as well? ^^

File: utilities/src/test_data.rs
Comment (wulfraem) [inline utilities/src/test_data.rs:1]: Hm, can the `SUBJECT_DID` now be removed from the test data?

File: tests/workflows.rs [hunk 3]
```diff
@@ -556,16 +529,13 @@ async fn workflow_can_create_finished_credential_without_credential_status(
 ) -> Result<(), Box<dyn Error>> {
     let mut vade = get_vade();
 
-    let revocation_list = create_revocation_list(&mut vade).await?;
-
-    let proposal = create_credential_proposal(&mut vade).await?;
+    let _revocation_list = create_revocation_list(&mut vade).await?;
```
Comment (wulfraem) [inline tests/workflows.rs:532]: Do we need the `_revocation_list` here?

**Review by wulfraem — CHANGES_REQUESTED** [review]

**Review by Tschuck — APPROVED** [review]

File: src/application/issuer.rs (hunk 1 above)
Comment (vineetpant) [reply src/application/issuer.rs]: removed, it was already present earlier and throwing warnings i just fixed warnings earlier but removing it more sense :)

File: src/application/prover.rs (hunk 2 above)
Comment (vineetpant) [reply src/application/prover.rs:59]: done

File: utilities/src/test_data.rs
Comment (vineetpant) [reply utilities/src/test_data.rs:1]: removed

File: tests/workflows.rs (hunk 3 above)
Comment (vineetpant) [reply tests/workflows.rs]: removed

**Comment by vineetpant** [comment]
Resolved conflicts and fixed the tests as per new changes in data types

**Review by S3bb1 — APPROVED** [review]

**Pull request merged**